packages/gasket-plugin-logger/lib/index.js

```diff
@@ -23,24 +23,28 @@ function verifyLoggerLevels(logger) {
 module.exports = {
   name,
   hooks: {
-    async init(gasket) {
-      const loggers = await gasket.exec('createLogger');
-      if (!loggers || loggers.length === 0) {
-        gasket.logger = {
-          debug: console.debug,
-          error: console.error,
-          info: console.info,
-          warn: console.warn,
-          child: (meta) => createChildLogger(this, meta)
-        };
-      } else if (loggers.length > 1) {
-        throw new Error(
-          'Multiple plugins are hooking createLogger. Only one logger is supported.'
-        );
-      } else {
-        verifyLoggerLevels(loggers[0]);
-        gasket.logger = loggers[0];
-      }
+    actions(gasket) {
+      return {
+        createLogger: async function () {
```

Comment:
Not crazy about this. We shouldn't have to call this at every entry.

Maybe setting up the logger is a case for keeping around `init`, but making it synchronous.

Alternatively, we make the action return the logger for an on demand approach.

```
myHook(gasket) {
  gasket.actions.getLogger().debug('yo')
}
```

Otherwise, let's attach as before during `init`.

```
myHook(gasket) {
  gasket.logger.debug('yo')
}
```


Current lifecycle flow is

1. `configure`
2. `actions`

So, if we want the logger to be available for use in `configure`, then that suggest `init` will still be important.

1. `init`
2. `configure`
3. `actions`

Or else, we register actions before configure, to make a getLogger action available

1. `actions`
2. `configure`

Reply:
I'm leaning towards keeping `init`. As much as I'd prefer to avoid attaching a bunch of stuff to the `gasket` instance, it is less of a code change across existing plugins for now.

Regardless, I'm trying to think how/if config would affect what actions are registered, and I don't that scenario exists. Sure, config make affect how an action behaves, but they should be predicably registered otherwise that's going to be a code and TS types nightmare.

So I'm going to propose we have

1. `init`
2. `actions`
3. `configure`

Let's leave `gasket.logger` for v7 and if we feel like we want be strict with actions instead of attachment, then we deprecate `gasket.logger` and move to `gasket.actions.getLogger()` for v8.

Reply:
Adding init and changing the order in https://github.com/godaddy/gasket/pull/728